packages/core/src/transport/WsOutboundTransport.ts

```diff
@@ -47,7 +47,7 @@ export class WsOutboundTransport implements OutboundTransport {
       throw new AriesFrameworkError("Missing connection or endpoint. I don't know how and where to send the message.")
     }
 
-    const isNewSocket = this.hasOpenSocket(endpoint)
+    const isNewSocket = !this.hasOpenSocket(endpoint)
     const socket = await this.resolveSocket({ socketId: endpoint, endpoint, connectionId })
```

Comment:
Comes down to, is return routing for a session until a message is sent without return routing, or is return routing enabled for a session if the first message has return routing